pkg/datapath/iptables/iptables.go

```diff
@@ -778,14 +784,40 @@ func (m *IptablesManager) endpointNoTrackRules(prog iptablesInterface, cmd strin
 	}
 	if err := prog.runProg([]string{
 		"-t", "filter",
-		cmd, ciliumInputChain,
+		cmd, ciliumForwardChain,
 		"-p", protocol,
 		"-d", IP,
 		"--dport",
 		p, "-j",
 		"ACCEPT"}); err != nil {
 		return err
 	}
+
+	// The following 2 rules cover packets from node-local-dns to
+	// non-host pod
+	if err := prog.runProg([]string{
```

Comment:
This might cause an issue during upgrade. Attempting to delete this rule when it doesn't exist would throw an error. That would prevent the other rules from getting cleaned up. Would it be better to aggregate all the errors instead of returning early?

Reply:
Hmm,  maybe I'm missing something but why would upgrade trying to delete these rules? The lifecycle of these rules is tied to that of the nodelocaldns pods.

Reply:
The sequence I was thinking of was:
- nodelocaldns pod is created, which causes Cilium to create the old iptables rules.
- Cilium is upgraded to a version which includes this change.
- nodelocaldns pod is deleted, which causes Cilium to delete the new iptables rules.
  - The new rules don't exist, so `endpointNoTrackRules` returns early with an error and doesn't clean up the rest of the old iptables rules.

I'm not sure if this is likely to happen though.

Reply:
^^ +1 
Also, why return early in case of an error? I tracked the call chain, and it looks like the errors are mainly logged.

Reply:
Makes sense, fixed :)

Reply:
> Cilium is upgraded to a version which includes this change.

This will repopulate iptables for nodelocaldns pods, right? otherwise, this change is not fixing existing clusters.

Reply:
Hmm, the trigger for repopulating is `UpdateNoTrackRules`, which happens in 3 cases:

- createEndpoint
- k8s pod no-track annotation update
- validateEndpoint

IIUC, when doing an in-place cilium upgrade, the new agent will restore endpoints which will trigger `validateEndpoint` and repopulate the rules.